### PR TITLE
test: add AKS integration test

### PR DESCRIPTION
**Pull request by dkontorovskyy**
- [ ] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [ ] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [X] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does
Add AKS integration test

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,6 +32,35 @@ commands:
...
+                command: mkdir -p /tmp/logs/test/integration/aks
+                name: Create temp dir for logs
+            - run:
+                command: |
+                    export NVM_DIR="/opt/circleci/.nvm"
+                    [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+                    nvm install v16
+                    npm install
+                    export KUBERNETES_MONITOR_IMAGE_NAME_AND_TAG=$(./scripts/circleci-jobs/setup-integration-tests.py)
+                    .circleci/do-exclusively --job ${CIRCLE_JOB} npm run test:integration:aks:yaml
```
Comment (ivanstanev) [inline .circleci/config.yml:54]: Is it OK to block on other PRs? Should we make it like the existing EKS and OS4 tests - run only on merge to staging?

File: test/README.md [hunk 2]
```diff
@@ -49,7 +50,7 @@ All integration tests require the Kubernetes-Monitor to be built into an image o
 The easiest way to achieve it is by running the `scripts/docker/build-image.sh` script.
 Please note that `docker` needs to be installed in order for this script to succeed.
 
-As part of these tests, we attempt pulling and scanning an image hosted on a private GCR registry. For this test case to work, one has to define the following environment variables: `GCR_IO_SERVICE_ACCOUNT`, `GCR_IO_DOCKERCFG`, `DOCKER_HUB_RO_USERNAME`, `DOCKER_HUB_RO_PASSWORD`.
+As part of these tests, we attempt pulling and scanning an image hosted on a private GCR registry. For this test case to work, one has to define the following environment variables: `GCR_IO_SERVICE_ACCOUNT`, `GCR_IO_DOCKERCFG`, `PRIVATE_REGISTRIES_DOCKERCFG`, `DOCKER_HUB_RO_USERNAME`, `DOCKER_HUB_RO_PASSWORD`.
```
Comment (ivanstanev) [inline test/README.md:53]: Do we still need `GCR_IO_DOCKERCFG`?

File: test/README.md [hunk 3]
```diff
@@ -84,6 +85,19 @@ This test runs whenever we commit to our `staging` branch, and at the moment may
 
 Run with `npm run test:integration:eks`.
 
+### AKS ###
+
+AKS is Azure's Kubernetes platform and helps us ensure we support not only the generic Kubernetes API, but also specifically AKS.
+
+This test uses an existing Azure account with an existing AKS cluster, and as such has a few more prerequisites:
+- `pip` is used to ensure the `az` CLI is installed and up to date. `az` is then used to generate a `kubeconfig` file to access the AKS cluster, as well as credentials to ACR.
+- AZ environment variables: `AZ_SP_APP_ID`, `AZ_SP_TENANT`, `AZ_SP_PASSWORD`, `AZ_ACR_REGISTRY`, `AZ_SUBSCRIPTION`, `AZ_RESOURCE_NAME`, `AZ_RESOURCE_GROUP` are used to authenticate against the Amazon account.
```
Comment (ivanstanev) [inline test/README.md:94]: `Amazon` -> `Azure`

File: package.json [hunk 4]
```diff
@@ -20,6 +20,7 @@
     "test:integration:kind:proxy": "DEPLOYMENT_TYPE=Proxy TEST_PLATFORM=kind CREATE_CLUSTER=true jest --logHeapUsage --ci --maxWorkers=1 test/integration/kubernetes.spec.ts",
     "test:integration:kindolm:operator": "DEPLOYMENT_TYPE=OperatorOLM TEST_PLATFORM=kindolm CREATE_CLUSTER=true jest --logHeapUsage --ci --maxWorkers=1 test/integration/kubernetes.spec.ts",
     "test:integration:eks:yaml": "DEPLOYMENT_TYPE=YAML TEST_PLATFORM=eks CREATE_CLUSTER=false jest --logHeapUsage --ci --maxWorkers=1 test/integration/kubernetes.spec.ts",
+    "test:integration:aks:yaml": "DEPLOYMENT_TYPE=YAML TEST_PLATFORM=aks CREATE_CLUSTER=false jest --logHeapUsage --ci --maxWorkers=1 test/integration/kubernetes.spec.ts",
```
Comment (ivanstanev) [inline package.json:23]: Side note: I think we should add a task to eventually deploy only with Helm in these tests and to deprecate the YAML installation

File: test/README.md [hunk 5]
```diff
@@ -84,6 +85,19 @@ This test runs whenever we commit to our `staging` branch, and at the moment may
...
+- `docker` is used to push the Kubernetes-Monitor's image to ACR.
+
+This test runs whenever we commit to our `staging` branch, and at the moment may only run once concurrently since it uses the same cluster.
+
+Run with `npm run test:integration:aks`.
```
Comment (ivanstanev) [inline test/README.md:99]: `test:integration:aks:yaml`

**Review by ivanstanev — APPROVED** [review]

File: test/README.md (hunk 2 above)
Comment (dkontorovskyy) [reply test/README.md:53]: Let it be there for now. Might be used for GCR specific tests

File: .circleci/config.yml (hunk 1 above)
Comment (dkontorovskyy) [reply .circleci/config.yml]: Fixed :) I've changed it for debug purpose and forgot 🤦

**Pull request merged**